### DISCUSSION TITLE
Traces across streams

### DISCUSSION THREAD

**Discussion by ngiagtzogloupsw** [openobserve/openobserve#4127] [Q&A] [answered]
Hello all,

If in a service ecosystem each service had its own designated stream-name would it be possible to have a combined view that would correlate all spans across services per trace_Id, or is this out of the question?

I noticed that in the UI Trace tab, we are able to select one stream at a time. And when expanding for a trace_Id we can only see the spans of this specific service and not others who might have spans for the same trace_id.

As of now I understand that all traces from all services participating in a request need to be in the same stream to be able to view the whole picture. Is this correct? And if yes, will there be any plans to support cross stream trace_id correlation?

Thank you

**Comment by prabhatsharma** [comment, marked as answer]
You should send all the related service to a single stream. There are no plans to support cross stream trace_id correlation
